sklearn/naive_bayes.py

```diff
@@ -680,10 +684,21 @@ class MultinomialNB(BaseDiscreteNB):
     """
 
     def __init__(self, alpha=1.0, fit_prior=True, class_prior=None):
-        self.alpha = alpha
+        self._alpha = alpha
```

Comment:
this violates scikit-learn API. Please check the setting of ``alpha`` in ``fit``. Also, only clip it when it is used, and don't change the value of ``self.alpha``.